libraries/chain/controller.cpp

```diff
@@ -1606,9 +1616,8 @@ struct controller_impl {
 
       emit( self.block_start, head->block_num + 1 );
 
-      auto guard_pending = fc::make_scoped_exit([this, head_block_num=head->block_num](){
-         protocol_features.popped_blocks_to( head_block_num );
-         pending.reset();
+      auto guard_pending = fc::make_scoped_exit([this](){
+         abort_block();
```

Comment:
Should not call potentially throwing code in a destructor.

Reply:
abort_block is not any more likely to throw than what was there before, but a try-catch couldn't hurt